commit a586dcc57ab35a269f0c271756951d49f422662d
Author: Ramana Raja <rraja@redhat.com>
Date:   Wed Feb 15 13:41:00 2023 -0500

    mgr: take a lock within PyModuleRegistry's mutators and accessor
    
    ... that modify and access the data member 'clients' respectively.
    
    Signed-off-by: Ramana Raja <rraja@redhat.com>

diff --git a/src/mgr/ActivePyModules.cc b/src/mgr/ActivePyModules.cc
index 02226270beb..8359a836963 100644
--- a/src/mgr/ActivePyModules.cc
+++ b/src/mgr/ActivePyModules.cc
@@ -1500,8 +1500,6 @@ void ActivePyModules::cluster_log(const std::string &channel, clog_type prio,
 
 void ActivePyModules::register_client(std::string_view name, std::string addrs)
 {
-  std::lock_guard l(lock);
-
   entity_addrvec_t addrv;
   addrv.parse(addrs.data());
 
@@ -1511,8 +1509,6 @@ void ActivePyModules::register_client(std::string_view name, std::string addrs)
 
 void ActivePyModules::unregister_client(std::string_view name, std::string addrs)
 {
-  std::lock_guard l(lock);
-
   entity_addrvec_t addrv;
   addrv.parse(addrs.data());
 
